lib/polymorphic_integer_type/extensions.rb

```diff
@@ -97,7 +101,28 @@ def has_one(name, scope = nil, **options)
         end
 
         remove_type_and_establish_mapping(name, options, scope)
-        super(name, options.delete(:scope), **options)
+        super(name, options.delete(:scope), **options.except(:foreign_integer_type, :integer_type)).tap do |_|
+          remove_integer_type_and_set_attributes_and_extension(options, ActiveRecord::Reflection::HasOneReflection, reflections[name.to_s])
+        end
+      end
+
+      def remove_integer_type_and_set_attributes_and_extension(options, klass, reflection)
+        foreign_integer_type = options.delete :foreign_integer_type
+        integer_type = options.delete :integer_type
+        is_polymorphic_integer = foreign_integer_type && integer_type
+
+        if is_polymorphic_integer
+          klass.attr_accessor(:foreign_integer_type)
+          klass.attr_accessor(:integer_type)
```

Comment:
Do we need to set two fields, or can we store the two pieces we need on a single attirubte?

Reply:
we do, cause we need to foreign_type and the actual assign value.

Reply:
I can store two pieces information on a single attribute, but that's usually not how reflection store their attributes which it feels awkward to me.

Reply:
I can live with 2 attributes.